### DISCUSSION TITLE
Dynamically Adding Context Modal Components to Modals Manager After User Enters Dashboard

### DISCUSSION THREAD

**Discussion by Sajarin-M** [mantinedev/mantine#3310] [Q&A]
Is there a way to dynamically add context modal components to the `modals manager` after the user has entered a specific dashboard, without specifying them in the `modals` prop of the root `ModalsProvider` component? I have three code-split dashboards in my app that all use the root `ModalsProvider` component, but each dashboard has different context modals that I would like to add dynamically.

**Comment by rtivital** [comment]
I do not think that it is a good idea to implmenent that

**Reply by Sajarin-M** [reply]
Thank you for your response. Could you please elaborate on why you think it is not a good idea? I am interested in learning more about the potential drawbacks or challenges that you foresee with this approach. Is there another way that you recommend handling the addition of context modal components in this situation?

**Reply by rtivital** [reply]
I do not recommend using modals package for this use case, just create your modals with Modal component if you want codesplitting to work.
